libseccomp/src/lib.rs

```diff
@@ -243,15 +245,15 @@ impl ScmpAction {
             Self::KillThread => SCMP_ACT_KILL_THREAD,
             Self::Trap => SCMP_ACT_TRAP,
             Self::Notify => SCMP_ACT_NOTIFY,
-            Self::Errno(x) => SCMP_ACT_ERRNO(x),
+            Self::Errno(x) => SCMP_ACT_ERRNO(x as u16),
             Self::Trace(x) => SCMP_ACT_TRACE(x),
             Self::Log => SCMP_ACT_LOG,
             Self::Allow => SCMP_ACT_ALLOW,
         }
     }
 
     /// Convert string seccomp action to ScmpAction
-    pub fn from_str(action: &str, errno: Option<u32>) -> Result<Self> {
+    pub fn from_str(action: &str, errno: Option<c_int>, msg_num: Option<u16>) -> Result<Self> {
```

Comment:
I think It would be better to just modify the argument name from `errno` to `val` or a similar name as a generic name in order to avoid changing the interface.

Example:
```rust
pub fn from_str(action: &str, val: Option<i32>) -> Result<Self> {
<...>
    "SCMP_ACT_ERRNO" => match val {
         Some(v) => Ok(Self::Errno(v)),
         None => Err(SeccompError::new(ParseError)),
    },
    "SCMP_ACT_TRACE" => match val {
         Some(v) => Ok(Self::Trace(v as u16)),
         None => Err(SeccompError::new(ParseError)),
    },
}
```